Duplicati/Library/Backend/TencentCOS/COSBackend.cs

```diff
@@ -0,0 +1,431 @@
+﻿using COSXML;
+using COSXML.Auth;
+using COSXML.Model.Bucket;
+using COSXML.Model.Object;
+using COSXML.Model.Tag;
+using COSXML.Utils;
+using Duplicati.Library.Common.IO;
+using Duplicati.Library.Interface;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Duplicati.Library.Backend
+{
+    /// <summary>
+    /// Tencent COS
+    /// https://cloud.tencent.com/document/product/436
+    /// https://cloud.tencent.com/document/product/436/32869
+    /// </summary>
+    public class COS : IBackend, IStreamingBackend, IRenameEnabledBackend
+    {
+        private static readonly string LOGTAG = Logging.Log.LogTagFromType<COS>();
+
+        private const string COS_APP_ID = "cos-app-id";
+        private const string COS_REGION = "cos-region";
+        private const string COS_SECRET_ID = "cos-secret-id";
+        private const string COS_SECRET_KEY = "cos-secret-key";
+        private const string COS_BUCKET = "cos-bucket";
+
+        private const bool COS_USE_SSL = true;
+        private const int CONNECTION_TIMEOUT_MS = 1000 * 60 * 60;
+        private const int READ_WRITE_TIMEOUT_MS = 1000 * 60 * 60;
+        private const int KEY_DURATION_SECOND = 60 * 60;
+
+        private readonly CosOptions _cosOptions;
+        private readonly string m_prefix;
+
+        private CosXml _cosXml;
+
+        /// <summary>
+        /// 腾讯云对象存储配置
+        /// </summary>
+        public class CosOptions
+        {
+            /// <summary>
+            /// 腾讯云账户的账户标识
+            /// </summary>
+            public string Appid { get; set; }
+            /// <summary>
+            /// 云 API 密钥 SecretId
+            /// </summary>
+            public string SecretId { get; set; }
+            /// <summary>
+            /// 云 API 密钥 SecretKey
+            /// </summary>
+            public string SecretKey { get; set; }
+            /// <summary>
+            /// 存储桶地域 ap-guangzhou ap-hongkong
+            /// https://cloud.tencent.com/document/product/436/6224
+            /// </summary>
+            public string Region { get; set; }
+            /// <summary>
+            /// 存储桶，格式：BucketName-APPID
+            /// </summary>
+            public string Bucket { get; set; }
+            /// <summary>
+            /// 存储桶中路径或子文件夹
+            /// </summary>
+            public string Path { get; set; }
+        }
+
+        //static COS()
+        //{
+        //}
+
+        public COS()
+        {
+        }
+
+        public COS(string url, Dictionary<string, string> options)
+        {
+            _cosOptions = new CosOptions();
+
+            var uri = new Utility.Uri(url);
+            m_prefix = uri.HostAndPath?.Trim()?.Trim('/').Trim('/')?.Trim('\\')?.Trim('\\');
```

Comment:
What's the purpose of the repeated `Trim('/').Trim('/')` and `Trim('\\')?.Trim('\\')`?  Isn't a single call to `Trim` for each character sufficient?